ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java

```diff
@@ -293,6 +293,9 @@ Map<String, String> getToolExecuteParams(MLToolSpec toolSpec, Map<String, String
             StringSubstitutor substitutor = new StringSubstitutor(executeParams, "${parameters.", "}");
             input = substitutor.replace(input);
             executeParams.put("input", input);
+            // Remove the current input from params to avoid passing it to the following tools.
+            // Otherwise, it will override the default input of the following tools.
+            params.remove("input");
```

Comment:
Intuitively, the method getToolExecuteParams is getting tool input parameters from `params`. The method should be a read-only method on `params`